msm/src/prover.rs

```diff
@@ -236,20 +238,14 @@ where
         quotient
     };
 
-    //~ 1. commit (hiding) to the quotient polynomial $t$.
-    //
-    // Our constraints are at most degree d2. When divided by
-    // vanishing polynomial, we obtain t(X) of degree d1.
-    let expected_t_size = 1;
-    // Quotient commitment
+    //~ 1. commit to the quotient polynomial $t$.
     let t_comm = {
-        let num_chunks = 1;
-        let mut t_comm = srs.commit_non_hiding(&quotient_poly, num_chunks);
-        let dummies_n = expected_t_size - t_comm.elems.len();
-        for _ in 0..dummies_n {
-            t_comm.elems.push(G::zero());
-        }
-        t_comm
+        let num_chunks: usize = if max_degree == 1 {
+            1
+        } else {
+            (max_degree - 1) as usize
+        };
+        srs.commit_non_hiding(&quotient_poly, num_chunks)
```

Comment:
I'm not sure I understand chunking enough to say if it's a correct way to implement things... The code above with `dummies_n` was copied from mips demo, and it handled mvlookup somehow? Maybe not.

It feels reasonable, but I want to understand this better ideally.

Reply:
Constraints for logup will be added in a following PR.

Reply:
The only issue I see at the moment is constraints of degree 1.

Reply:
What I don't know here is if it should be non hiding or not. Maybe are you using non-hiding because they don't want zk?